easybuild/easyblocks/generic/condacreate.py

```diff
@@ -28,84 +28,66 @@
 """
 
 import os
-import shutil
-import stat
 
-import easybuild.tools.environment as env
-from easybuild.framework.easyblock import EasyBlock
+from easybuild.easyblocks.anaconda import initialize_conda_env, set_conda_env
+from easybuild.easyblocks.generic.binary import Binary
 from easybuild.framework.easyconfig import CUSTOM
-from easybuild.tools.build_log import EasyBuildError
-from easybuild.tools.filetools import rmtree2
 from easybuild.tools.run import run_cmd
-from easybuild.easyblocks.anaconda import post_install_step, pre_install_step, initialize_conda_env, set_conda_env
 
-class EB_CondaCreate(EasyBlock):
+
+class CondaCreate(Binary):
     """Support for building/installing CondaCreate."""
 
     @staticmethod
     def extra_options(extra_vars=None):
         """Extra easyconfig parameters specific to EB_CondaCreate easyblock."""
-        extra_vars = EasyBlock.extra_options(extra_vars)
+        extra_vars = Binary.extra_options(extra_vars)
         extra_vars.update({
-            'requirements': [None, "Requirements files", CUSTOM],
             'channels': [None, "Custom conda channels", CUSTOM],
             'post_install_cmd': [None, "Commands after install: pip install, cpanm install, etc", CUSTOM],
             'pre_install_cmd': [None, "Commands before install: setting environment variables, etc", CUSTOM],
+            'requirements': [None, "Requirements file", CUSTOM],
         })
         return extra_vars
 
-    def __init__(self, *args, **kwargs):
-        """Initialize EB_CondaCreate-specific variables."""
-        super(EB_CondaCreate, self).__init__(*args, **kwargs)
-
-
     def extract_step(self):
-        """Move all source files to the build directory"""
-
-        if not self.src:
-            pass
+        """No sources expected."""
+        if self.src:
+            super(EB_CondaCreate, self).extract_step()
         else:
-            self.src[0]['finalpath'] = self.builddir
-
-            # copy source to build dir.
-            for source in self.src:
-                src = source['path']
-                dst = os.path.join(self.builddir, source['name'])
-                try:
-                    shutil.copy2(src, self.builddir)
-                    os.chmod(dst, stat.S_IRWXU)
-                except (OSError, IOError), err:
-                    raise EasyBuildError("Couldn't copy %s to %s: %s", src, self.builddir, err)
-
-    def configure_step(self):
-        """No configuration, this is binary software"""
-        pass
-
-    def build_step(self):
-        """No compilation, this is binary software"""
-        pass
+            pass
 
     def install_step(self):
-        """Copy all files in build directory to the install directory"""
+        """Set up conda environment using 'conda create' and install using specified requirements."""
+
+        if self.cfg['pre_install_cmd']:
+            run_cmd(self.cfg['pre_install_cmd'], log_all=True, simple=True)
```

Comment:
Since this is a site-specific thing, why not make `eb` an alias that always unsets `$PERL5LIB`?

Doing this in each and every easyconfig doesn't seem like a good way to go, especially since there's an equivalent already via `preinstallopts`.